src/astprinter.nit

```diff
@@ -90,10 +175,66 @@ redef class ABlockExpr
 	end
 end
 
-redef class AIntegerExpr
-	redef fun accept_printer(v)
+redef class ADoExpr
+	redef fun number_of_lines do return n_block.number_of_lines
+	redef fun accept_printer(v) do
+		v.visit_block("do", n_block)
+	end
+end
+
+redef class ALoopExpr
+	redef fun number_of_lines do return n_block.number_of_lines
+	redef fun accept_printer(v) do
+		v.visit_block("loop", n_block)
+	end
+end
+
+redef class AForExpr
+	redef fun number_of_lines do return n_block.number_of_lines
+	redef fun accept_printer(v) do
+		v.write("for ")
+		for g in n_groups do
+			v.write(v.varname(g.variables.first))
+			v.write(" in ")
+			v.visit(g.n_expr)
+		end
+		v.visit_block("do", n_block)
+	end
+end
+
+redef class AIfExpr
+	redef fun number_of_lines
 	do
-		v.write(value.to_s)
+		 #n_then.number_of_lines
+		 #n_else.number_of_lines
+		 return 3
```

Comment:
Cleanup of comments?
Refactor magic number 3 to show explicit use of visit_block_noend else condition.